### PR TITLE
[Deprecated] - derenma 6/12

### PR DESCRIPTION
**Pull request by endian0A**
###### Motivation for this change

Strange dependency issues when bundling some Gems.  Fixed with updates.
###### Things done
- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [X] NixOS
  - [ ] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

…ndling some

gems

### CODE REVIEW

**Comment by joachifm** [comment]
For the benefit of reviewers: please use descriptive PR titles and a clean history, such that it becomes clear what the proposal is intended to achieve and how it does it.

File: pkgs/tools/security/metasploit/default.nix [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
 stdenv.mkDerivation rec {
   name = "metasploit-framework-${version}";
   version = "4.12.5";
 
 
-  src = fetchgit {
-    url = "https://github.com/rapid7/metasploit-framework.git";
-    sha256 = "0rzn2lax3nvhgmfivr7jdifir9j8f1f7mk2ypksr8lvqk4hr54kw";
+  src = fetchurl {
+    url = "https://github.com/rapid7/metasploit-framework/archive/4.12.5.tar.gz";
```
Comment (zimbatm) [inline pkgs/tools/security/metasploit/default.nix]: it's better to use `fetchFromGitHub { owner = "rapid7"; repo = "metasploit-framework"; rev = "4.12.5"; sha256 = "..."; }`

File: pkgs/tools/security/metasploit/default.nix [hunk 2]
```diff
@@ -21,10 +21,9 @@ stdenv.mkDerivation rec {
 
     bundle install --path=$out/.gem
 
-
-    # !!! Must manually set bundler gem path !!!! 
-    export GEM_HOME=$out/.gem
-    export HOME=$out/.gem/bin:$PATH
+    # !!! This needs to be tweeked.
+    export GEM_HOME=~/gem/ruby/2.3.0/cache
```
Comment (zimbatm) [inline pkgs/tools/security/metasploit/default.nix]: this is now dependent on the ruby version. look at how other packages how they are using the `ruby.gemPath`

**Comment by endian0A** [comment]
Pull request closed.  I am such a git n00b, many apologies. I didn't realize that the same pull request would reflect commits to my fork.  I am reading more into git today to ensure that my changes are productive.

Bundler upgrade was due to a non-intuitive error messages when 'Gemfile' was not found in the relative path.  

Metasploit Update:
msfupdate 
- This app has a fixed requirement for bundler.  If ran on NixOS, it will attempt a 'git clone' and 'bundler install' inside of the current user profile, which is not the "NixOS way" of updating things.

- I will not link to dynamic ruby paths!  That was implemented during an earlier test to see if I could 'force' bundler to install itself and per earlier note, and NOT intended to be part of the pull.  Again, my mistake.

- These issues with msfupdate are best solved (in my opinion) by patching out (or not installing) msfupdate and maintaining a patch to resolve the dependency check for bundler.

**Comment by zimbatm** [comment]
No worries. Just look into `git rebase -i origin/master` and edit the commits to make them cleaner. The you can `git push -f` on the same branch.

* Each commit should be a logical change. It's almost there.
* Commit titles should not go past 78 characters. To add more context, skip a line and put further text below.
* If your commit affects a package, start the title with the package name and a color (eg: `metasploit:`)

**Comment by zimbatm** [comment]
Eg: `bundler: 1.11.2 -> 1.11.5\n\nresolves bizarre dependency issues when building metasploit`

**Comment by endian0A** [comment]
Closing THIS pull request, as I nuked my original fork.

Rebuilt my dev environment from scratch, rebuilt my fork, and will be much more cognizant of what is included in my pull requests. 

Until I get the hang of group development, I will be focusing on the Metasploit package ONLY just to keep things simple as I get more experience.

Thanks for being patient with a n00b.  :)

**Comment by zimbatm** [comment]
No worries. Ping me on the other PRs if you need any help.

**Pull request closed without merging**